src/inmanta/protocol/rest/client.py

```diff
@@ -40,14 +40,18 @@ class RESTClient(RESTBase):
     HTTP verbs. For other methods the POST verb is used.
     """
 
-    def __init__(self, endpoint: "Endpoint", connection_timout: int = 120) -> None:
+    def __init__(self, endpoint: "Endpoint", connection_timout: int = 120, force_instance: bool = False) -> None:
         super().__init__()
         self.__end_point: "Endpoint" = endpoint
         self.daemon: bool = True
         self.token: Optional[str] = inmanta_config.Config.get(self.id, "token", None)
         self.connection_timout: int = connection_timout
         self.headers: Set[str] = set()
         self.request_timeout: int = inmanta_config.Config.get(self.id, "request_timeout", 120)
+        max_clients: int = inmanta_config.Config.get(
+            "agent_rest_transport", "max_clients", 1000
+        )  # todo what should be the default here?
```

Comment:
Wouldn't it be a lot easier to use `AsyncHTTPClient.configure` when the agent starts? I assume there's a reason for the singleton behavior tornado uses, and we actually want to apply it agent-wide anyway. It would also fix the bug(?) you have now where the `agent_rest_transport.max_clients` setting is used for any rest client, not only the agent's clients.

Reply:
I don't find how to make AsyncHTTPClient.configure work but I fixed the fact that is was set for any rest client

Reply:
I'm going to look into it because I would if we can.